sys/net/network_layer/ipv6/addr/ipv6_addr_from_str.c

```diff
@@ -49,7 +51,8 @@ ipv6_addr_t *ipv6_addr_from_str(ipv6_addr_t *result, const char *addr)
     uint8_t saw_xdigit = 0;
     uint8_t i = 0;
 
-    if ((result == NULL) || (addr == NULL)) {
+    if ((result == NULL) || (addr == NULL) || (addr_len == 0) ||
+        (addr_len > IPV6_ADDR_MAX_STR_LEN)) {
```

Comment:
I guess `IPV6_ADDR_MAX_STR_LEN` does not account for `\0`?  This line probably has to check for `IPV6_ADDR_MAX_STR_LEN + 1`, but I am not sure.  Maybe you can also extend the test cases below with strings that hit the max length?

Reply:
On a second thought, including the length for `\0` is probably not necessary anymore with the proposed changes.  But having a test case for that is still helpful.

Reply:
It [does actually account for the `\0`](https://github.com/RIOT-OS/RIOT/blob/d85f740df3ab2ab83a539446a4233e89bcb7d714/sys/include/net/ipv6/addr.h#L45-L55).

Reply:
Added a test case